packages/translated-components/src/index.js

```diff
@@ -61,17 +62,33 @@ const moneyFormat = (language) => ({
 
 const preHeat = (translations, format) => (
   reduce(translations, (acc, o, language) => {
-    acc[language] = reduce(o, (acc, v, k) => {
-      acc[k] = new IntlFormat(v, kebabCase(language), {...moneyFormat(language), ...format})
-      return acc
-    }, {})
+    acc[language] = preheatValues(o, format, language)
     return acc
   }, {})
 )
 
-const templateReducer = (values) => (acc, v, k) => {
-  acc[k] = v.format(values)
-  return acc
+const preheatValues = (obj, format, language) => {
+  return reduce(obj, (acc, value, key) => {
+    if (typeof value === 'object') {
+      acc[key] = preheatValues(value, format, language)
+    } else {
+      acc[key] = new IntlFormat(value, kebabCase(language), {...moneyFormat(language), ...format})
+    }
+    return acc
+  }, {})
+}
+
+const templateReducer = (values) => partial(templateIteratee, values)
```

Comment:
I'm personally inclined to avoid another dependency when all it takes is just returning a function after the first arg. I'm not too fussed though so happy to defer.